tex/s2c.tex

```diff
@@ -66,9 +68,35 @@ \subsection{Security}
 \end{figure}
 
 \begin{definition}[Commitment Binding for Sign-to-Contract]\label{def:euf-cma-s2c}
+	A sign-to-contract signature scheme $\algo{SigS2C}$ is \emph{strongly binding} if for all \ppt adversaries $\adv$:
+	\[
+	  \advantage{ComStrongBind}{\adv, \algo{SigS2C}} \defeq \pr{\game{\Game~\algo{ComStrongBind }}{\algo{SigS2C}} = 1} = \negl
+	\]
+	where $\game{\Game~\algo{ComStrongBind}}{\algo{SigS2C}}$ is defined in \autoref{fig:strongly-binding-s2c}.
 \end{definition}
```

Comment:
Maybe we can define a `Commit` wrapper that allows us to reuse all our commitment defintions?
We'd say somewhere `Commit(pp, m, r) = S2CSign(\params, sk, m_0, m, r)` but then it's not clear where `sk` comes from.

Hm, here's a somewhat different approach to this PR that uses the fact that the commitment is really not the entire signature but just its "R-value":
- Add a Commit algorithm: `Commit(pp, m, R_C) -> R`
- Change VerifyCom to `VerifyS2C(pp, \sigma, R)`, which in the case of Schnorr signatures just checks that `\sigma` contains `R`.

Then `(S2CSign.Setup, S2CSign.Commit)` is a commitment scheme and we can use all existing definitions for commitment schemes (well we don't have one for strong binding, except in the solutions).